index.ts

```diff
@@ -167,6 +167,16 @@ function isNodeKindExportAssignment(value: ts.Node): value is ts.ExportAssignmen
 
 export default function generate(options: Options): Promise<void> {
 
+  // validate options
+	if (options.name && options.main) {
+		if (options.main.indexOf(options.name + '/') !== 0) {
```

Comment:
sure.  My thinking was that the meaning of `options.main` seemed a little underspecified and from the usage in the script it appears to be the name of the module we want to "export" as <name>.  So this makes a stab at enforcing that the exported main module is one of the ones that we actually generate a definition for.  it's not perfect but without any checking it's unclear if main should be a path, a module name (before our prefixing of <name>) or a module name after prefixing <name>.  Usage suggests that the last is intended.

I'm open to ideas of how to better enforce this, as this is fairly rudimentary (compared to, say, making sure we actually process a module of name `options.main`... and now that i'm putting it this way I'm starting to think that an assertion of that type would make more sense)